File: BACKEND/src/main/java/com/model/Clinic.java
package com.model;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Set;

@Entity
public class Clinic {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(columnDefinition = "VARCHAR(50)",nullable = false)
    private String name;
    @Column(nullable = false)
    private String address;
    @Column(nullable = false)
    private String description;

    @OneToMany(mappedBy = "clinic", fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private Set<Examination> examinations=new HashSet<>();


    @OneToMany(mappedBy = "clinic", fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private Set<Doctor> doctors=new HashSet<>();

    @OneToMany(mappedBy = "clinic", fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private Set<Nurse> nurses=new HashSet<>();

    @OneToMany(mappedBy = "clinic", fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private Set<Room> rooms=new HashSet<>();

    @OneToMany(mappedBy = "clinic", fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private Set<ClinicAdministrator> clinicAdministrators=new HashSet<>();

    @OneToMany(mappedBy = "clinic", fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private Set<ExaminationType> examinationTypes=new HashSet<>();



    public Clinic() {
    }

}

File: FRONTEND/src/app/models/clinic.ts
import { Doctor } from './doctor';
import { ExaminationType } from './examinationType';

export class Clinic {
  id: number;
  name: string;
  address: string;
  description: string;
  clinicRating: number;
  doctors: Array<Doctor> = new Array<Doctor>();
  examinationTypes: Array<ExaminationType> = new Array<ExaminationType>();

  constructor(name: string, address: string, description: string, doctors: Array<Doctor>, examinationTypes?: Array<ExaminationType>, clinicRating?: number, id?: number) {
    this.name = name;
    this.address = address;
    this.description = description;
    this.doctors = doctors;
    this.clinicRating = clinicRating;
    this.examinationTypes = examinationTypes;
    this.id = id;
  }
}